### PR TITLE
API changed, "processFrameFn" option provided, version bumped

### PR DESCRIPTION
**Pull request by georgiy-abbasov**
\cc @inikulin

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,7 +88,24 @@ const createCallsiteRecord = require('callsite-record');
...
         (function func3() {
-            const record = createCallsiteRecord('func2');
+            const record = createCallsiteRecord({ byFunctionName: 'func2' });
+        })();
+    })();
+})();
+```
+
+You can specify `processFrameFn` function, which will process every frame in callstack. It's usefull when you need to 
+enable frame processing like `source-maps-support` or something other.
```
Comment (inikulin) [inline README.md]: `or something other` - you can remove this part.

File: lib/index.js [hunk 2]
```diff
@@ -282,17 +282,23 @@ CallsiteRecord.fromError = function (error, isCallsiteFrame) {
...
+module.exports = function createCallsiteRecord (options) { /*{ forError, isCallsiteFrame, byFunctionName, typeName, processFrameFn }*/
+    if (options.forError)
+        return CallsiteRecord.fromError(options.forError, options.isCallsiteFrame);
 
-    else if (typeof arguments[0] === 'string') {
+    else if (options.byFunctionName) {
         var stackFrames = callsite();
 
+        if (options.processFrameFn) {
+            stackFrames = stackFrames.map(function (frame) {
```
Comment (inikulin) [inline lib/index.js]: I guess this should be done in `CallsiteRecord` ctor, since you may want to do that for error as well.

Comment (georgiy-abbasov) [inline lib/index.js]: I agree, that this ability should be available, but processing should be executed before `CallsiteRecord` ctor

**Comment by georgiy-abbasov** [comment]
FPR

File: lib/index.js [hunk 3]
```diff
@@ -74,6 +74,12 @@ function getFrameMethodName (frame, funcName) {
     return typeName + '.' + (methodName || '<anonymous>');
 }
 
+function processFrames (stackFrames, processFn) {
+    return stackFrames.map(function (frame) {
+        return processFn(frame);
+    });
```
Comment (inikulin) [inline lib/index.js]: It can be `stackFrames.map(processFn)`. Move `processFn` check in this function.

**Comment by georgiy-abbasov** [comment]
FPR

**Comment by georgiy-abbasov** [comment]
FPR

**Pull request merged**